imperial.js

```diff
@@ -112,13 +106,16 @@ export default class Imperial {
   }
 
   tick(action) {
-    if (action.type === "playerSeating") {
-      this.seatPlayers(action);
-    } else if (action.type === "assignStartingNation") {
-      this.assignStartingNation(action);
+    this.log.push(action);
+
+    if (action.type === "init") {
+      Object.assign(this, setup(action.payload));
```

Comment:
Explicitly use `this.players = ...` and `this.nations = ...` etc.




pl00z